src/App.js

```diff
@@ -22,34 +22,58 @@ export function App () {
         FILE_VIEW: 1,
     };
 
-    const urlSearchParams = new URLSearchParams(window.location.search, "?");
-
     const [fileInfo, setFileInfo] = useState(null);
     const [theme, setTheme] = useState(THEME_STATES.DARK);
     const [appMode, setAppMode] = useState(null);
+    const [prettify, setPrettify] = useState(null);
+    const [logEventIdx, setLogEventIdx] = useState(null);
 
     const switchTheme = (theme) => {
         localStorage.setItem("ui-theme", theme);
         document.getElementById("app").setAttribute("data-theme", theme);
         setTheme(theme);
     };
 
-    useEffect(() => {
-        console.debug("Version:", config.version);
-        const lsTheme = localStorage.getItem("ui-theme");
-        switchTheme(lsTheme === THEME_STATES.LIGHT?THEME_STATES.LIGHT:THEME_STATES.DARK);
+    /**
+     * Loads the initial state of the viewer from url arguments.
+     *
+     * ### Example:
+     * localhost:3010/?filePath=/logs/sample.clp.zst&prettify=true&logEventIdx=1
+     *
+     * @return {[string,boolean,string]} [filePath,prettify,logEventIdx]
```

Comment:
Why not convert logEventIdx into a number in this method?

Reply:
I validate that it is a number in the viewer before I cast it into a number. This is done because in the URL, they could enter a string for logEventIdx.